File: learning-akka/util/src/main/kotlin/book/kotlin/messages/Messages.kt
package book.kotlin.messages

import java.io.Serializable

data class SetRequest(val key: String, val value: Any) : Serializable

data class SetIfNotExists(val key: String, val value: Any) : Serializable

data class GetRequest(val key: String) : Serializable

data class DeleteRequest(val key: String) : Serializable

data class KeyNotFoundException(val key: String) : Exception("Key $key not found in database"), Serializable

data class KeyAlreadyExistsException(val key: String) : Exception("Key $key already exists in database"), Serializable

File: learning-akka/util/src/main/scala/book/scala/messages/Messages.scala
package book.scala.messages

case class SetRequest(key: String, value: Any)

case class SetIfNotExists(key: String, value: Any)

case class GetRequest(key: String)

case class DeleteRequest(key: String)

case class KeyNotFoundException(key: String) extends Exception(s"Key $key not found in database")

case class KeyAlreadyExistsException(key: String) extends Exception(s"Key $key already exists in database")
